modules/core/src/main/java/org/apache/fluo/core/util/ScanUtil.java

```diff
@@ -89,68 +115,158 @@ public static Span getSpan(ScanOpts options) {
       try (Snapshot s = client.newSnapshot()) {
 
-        Span span = null;
-        Collection<Column> columns = null;
-        try {
-          span = getSpan(options);
-          columns = getColumns(options);
-        } catch (IllegalArgumentException e) {
-          System.err.println(e.getMessage());
-          System.exit(-1);
+        Span span = getSpan(options);
+        Collection<Column> columns = getColumns(options);
+
+        // Retrieve "fluo.scan.*" from config properties.
+        SimpleConfiguration scanProperties = sConfig.getScanConfiguration();
+        if (options.exportAsJson) {
+          generateJson(options, scanProperties, span, columns, s, out);
+        } else { // TSV or CSV format
+          generateTsvCsv(options, scanProperties, span, columns, s, out);
         }
 
-        CellScanner cellScanner = s.scanner().over(span).fetch(columns).build();
-
-        StringBuilder sb = new StringBuilder();
-        for (RowColumnValue rcv : cellScanner) {
-          if (options.hexEncNonAscii) {
-            sb.setLength(0);
-            Hex.encNonAscii(sb, rcv.getRow());
-            sb.append(" ");
-            Hex.encNonAscii(sb, rcv.getColumn(), " ");
-            sb.append("\t");
-            Hex.encNonAscii(sb, rcv.getValue());
-            System.out.println(sb.toString());
-          } else {
-            sb.setLength(0);
-            sb.append(rcv.getsRow());
-            sb.append(" ");
-            sb.append(rcv.getColumn());
-            sb.append("\t");
-            sb.append(rcv.getsValue());
-            System.out.println(sb.toString());
-          }
+      } catch (FluoException e) {
+        throw e;
+      }
+    }
+  }
 
-          if (System.out.checkError()) {
-            break;
-          }
+  /**
+   * Generate TSV or CSV format as result of the scan.
+   * 
+   * @since 1.2
+   */
+  private static void generateTsvCsv(ScanOpts options, SimpleConfiguration scan, Span span,
+      Collection<Column> columns, final Snapshot snapshot, PrintStream out) throws IOException {
+    // Default TAB separator and NO quotes if possible.
+    CSVFormat csvFormat = CSVFormat.DEFAULT;
+    csvFormat = csvFormat.withDelimiter(CSVFormat.TDF.getDelimiter());
+    csvFormat = csvFormat.withQuoteMode(QuoteMode.MINIMAL);
+
+    // when "--csv" parameter is passed the "fluo.scan.csv" is analised
+    if (options.exportAsCsv) {
+      if (scan.containsKey(CSV_DELIMITER)
+          && StringUtils.isNotEmpty(scan.getString(CSV_DELIMITER))) {
+        csvFormat = csvFormat.withDelimiter(scan.getString(CSV_DELIMITER).charAt(0));
+      }
+
+      if (scan.containsKey(CSV_ESCAPE) && StringUtils.isNotEmpty(scan.getString(CSV_ESCAPE))) {
+        csvFormat = csvFormat.withEscape(scan.getString(CSV_ESCAPE).charAt(0));
+      }
+
+      if (scan.containsKey(CSV_QUOTE) && StringUtils.isNotEmpty(scan.getString(CSV_QUOTE))) {
+        csvFormat = csvFormat.withQuote(scan.getString(CSV_QUOTE).charAt(0));
```

Comment:
When I saw this `charAt(0)` it made me think of having the sanity check to ensure option only has one char.

Reply:
I'll think a way to do this.